src/server.js

```diff
@@ -155,11 +166,13 @@ Server.prototype.receiveEdit = function(connection, editMessage, sendToClient){
         if(!isEmpty(edit.diff)){
           clientDoc.shadow.localVersion++;
         }
+
+        // notify about patch
+        this.emit('patch', { clientDoc, connection, edit, editMessage });
       }else{
         // TODO: implement backup workflow
         // has a low priority since `packets are not lost` - but don't quote me on that :P
-        console.log('error', 'patch rejected!!', edit.serverVersion, '->', clientDoc.shadow.serverVersion, ':',
-                    edit.localVersion, '->', clientDoc.shadow.localVersion);
+        this.emit('patch_rejected', { clientDoc, connection, edit, editMessage });
```

Comment:
I'm cool with bumping the major version since it seems that everybody is getting on the ES6 bandwagon. Up to you though.